lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -460,17 +507,35 @@ def handle_response(response)
       #   - a string or an array of strings (if one of many responses)
       #   - An array of strings if one of many responses could be considered a
       #     success.
```

Comment:
We need the ability to return success based on responses materially different than looking for `:ok` in the parsed response. The `:last_event` stuff also wouldn't work for `store` because the response for `store` doesn't include last event information. I broke things down into simpler methods for each type of success-check, and left things more easily extensible for any future types of action-specific success criteria we may need (rather than continue to pile on the existing conditional logic in the original method).

Reply:
I'd vote to just axe `success_and_message_from` step too.